### ISSUE TITLE
Several Issues on Latest Release

### ISSUE THREAD

**Issue by SyntaxDreamer** [RickStrahl/MarkdownMonster#1155]
I purchased a license without proper testing after just one or two hours. Maybe I should have as there are many bugs which I want to address:

1. Copying text to the clipboard does not work. Always shows an error. From the logs:
Couldn't retrieve image from Clipboard. 

But the status bar in the software seems more useful with:

**Comment by RickStrahl** [comment]
Where does the image originate?

Clipboard copy works - I use it all the time, but there are possible complications with certain clipboard formats that some programs use.  Is this something other than SnagIt?  

SnagIt captures from clipboard should definitely work - I heavily use SnagIt (and Windows Capture occasionally) myself.

Athough I admittedly just use SnagIt natively without the MM addin. 

I just checked the addin with SnagIt 2024 (I'd been using 2020) and that version fails for me as well - works with SnagIt 2020 though so this looks like something broke in SnagIt's COM object in the 2024 (or interim releases).

I'll take a look at it, but in the meantime this is the workflow that I use with SnagIt:

* Bring up SnagIt capture externally (hot keyed or from task tray)
* Capture 
* SnagIt Automatically captures to clipboard
* or... opens in editor and edit image, then copy image to clipboard
* Paste into MM

You can also use the add-in flow, if you just don't finish but manually copy to clipboard and cancel, then paste. The finsh operation in SnagIt is what's broken - this seems like it's a SNagIt issue, not an MM issue.

**Comment by SyntaxDreamer** [comment]
It only happens if the Preview option is enabled in the settings, the workflow in that scenario is as follows:

a) Snagit runs, you make the screenshot.
b) The image opens in the Snagit editor with a Finish button. This allows you to draw arrows or edit the image before.
c) Click Finish, then it crashes Snagit instead of returning to the app.

I tried copy from the Snagit editor and pasting it, it works. I guess then the Snagit issue is not critical, as you said, I can take the screenshot with my regular shortcut and use the paste feature which works equally well for images.

As for the error when copying in the app, this is not related to Snagit. I happen when you select some text inside the editor and want to copy it. You get this error:

![image](https://github.com/user-attachments/assets/8fba0ca2-2157-4fb2-a608-74437b7fc834)

Just text. Any text I select and try to copy (ctrl+c) as if the app is blocked from using the Windows clipboard.

This only happens in the working windows (where you type or edit text). You can copy text fine from the preview window, but not while writing or editing text in the actual markdown file.

**Comment by RickStrahl** [comment]
Do you have multiple versions of SnagIt installed? In the past I've seen this issue crop up when multiple versions are installed.

**Comment by RickStrahl** [comment]
Not sure about the clipboard issue. 

Any chance this is the issue:
https://stackoverflow.com/questions/68666/clipbrd-e-cant-open-error-when-setting-the-clipboard-from-net

**Comment by RickStrahl** [comment]
For the clipboard issue: There are multiple ways to get at the clipboard. I've added some code that tries both mechanisms now in case this isn't a system issue. 

You can try this in `3.5.5.7`.

It won't fix the issue referenced in the SO post above though - that's a WIndows system issue.

**Comment by RickStrahl** [comment]
Can you add a separate issue for the monitor startup with a bit more information please. What's your setup how are you restarting, etc.

**Comment by SyntaxDreamer** [comment]
> Do you have multiple versions of SnagIt installed? In the past I've seen this issue crop up when multiple versions are installed.

No, just one.

**Comment by SyntaxDreamer** [comment]
> Not sure about the clipboard issue.
> 
> Any chance this is the issue: https://stackoverflow.com/questions/68666/clipbrd-e-cant-open-error-when-setting-the-clipboard-from-net
> 
> Also I seem to remember that in some cases when running As Administrator can cause problem with the clipboard when copying from a non-admin process or vice versa.

Yes, that seems to be the issue. Only seems to affect the active editor window.

**Comment by RickStrahl** [comment]
Did you try the latest version that has some updated Clipboard logic (based on the SO post)...

**Comment by RickStrahl** [comment]
One more thing:

Can you check your Configuration file for the  `WindowPosition` value after you've closed MM in the

From a Command Prompt (or use Explorer to navigate):
```cmd
notepad "%appdata%\Markdown Monster\MarkdownMonster.json"
```

* Ctrl-F for `WindowPosition`
* Copy that section  `WindowPosition: { }`  and paste here (top/left/width/height/WindowState)

I suspect what's happening is that you have the MM window maximized which currently looks like it ignores the positioning. If that's so can you try for now not maximizing and then closing and reopening?

**Comment by SyntaxDreamer** [comment]
Sorry, was busy. Downloaded latest version 3.5.6.2.

Good news. The copy error is gone. Fixed.

The bad news. The windows/monitor position is now worse than before.

**Before:**
It was always launching the application on my main monitor.

After:
Now it's launching it split between 2 monitors, offset, only a little part of the bottom is visible, the other parts are offset from the view port. I just move them and maximize, hitting windows shortcut keys, but for someone who does not know the shortcuts on Windows they will be stuck without being able to select the application and drag it back with their cursor. This can be very frustrating for some people if an application goes offset, and it's not visible.

The way I open it:

I just open a markdown file, I set .md files to use Markdown Monster as default. In the settings I have the checkbox option enabled 'Open in Presentation Mode' because I mostly want to open files to read them first, as documentation by default. 

I decided to test this with the setting off. And then it seems to work. It seems to remember the monitor, and it opens the applications correctly. I then check the setting again, and it goes off-screen once more, between 2 monitors (none of which are the one I actually set before).

It seems to be isolated to when Presentation Mode is turned on.

The JSON after closing the app (with Presentation Mode checked)

```json
  "WindowPosition": {
    "Top": 1080,
    "Left": 516,
    "Width": 1050,
    "Height": 690,
    "InternalPreviewWidth": 883,
```

I think you just need to look into Presentation Mode, I will turn it off for now, not a big deal, but at least we know where the bug is 😊

**Comment by RickStrahl** [comment]
I moved the Window Position issue over to #1158.

Closing this issue since the main issues of the Clipboard seems to be fixed with the switch to using the Window Forms clipboard and delayed retrying on failure.
